rest/auth.py

```diff
@@ -311,6 +311,17 @@ def inner(*args, **kwargs):
 
     return decorator
 
+
+# Retrive current user role and login
+def get_current_user():
+    if not auth_enabled:
+        return abort(403, "Authentication has \
+            to be enabled to retrieve current user")
+
+    token = request.headers['Authorization'].split()[1]
```

Comment:
The token extraction logic should be factorized to avoid loosely access to request headers dict from anywhere in the code.